Detectors/MUON/MCH/Simulation/src/MCHDigitizer.cxx

```diff
@@ -0,0 +1,206 @@
+  //4) handling of MCtruth information
+}
+
+//______________________________________________________________________
+
+void MCHDigitizer::process(const std::vector<Hit> hits, std::vector<Digit>& digits)
+{
+  //array of MCH hits for a given simulated event
+  for (auto& hit : hits) {
+    processHit(hit, mEventTime);
+  } // end loop over hits
+  //TODO: merge (new member function, add charge) of digits that are on same pad:
+  //things to think about in terms of time costly
+
+  digits.clear();
+  fillOutputContainer(digits);
+}
+
+//______________________________________________________________________
+int MCHDigitizer::processHit(const Hit& hit, double event_time)
+{
+
+  //hit position(cm)
+  Point3D<float> pos(hit.GetX(), hit.GetY(), hit.GetZ());
+  //hit has global coordinates
+  //convert energy to charge, float enough?
+  float charge = mMuonresponse.etocharge(hit.GetEnergyLoss());
+  //std::cout << "charge " << charge << std::endl;
+
+  //time information
+  float time = hit.GetTime(); //to be used for pile-up
+  int detID = hit.GetDetectorID();
+  //get index for this detID
+  int indexID = mdetID.at(detID);
+  //# digits for hit
+  int ndigits = 0;
+
+  //transformation from global to local
+  auto t = o2::mch::getTransformation(detID, *gGeoManager);
+  Point3D<float> lpos;
+  t.MasterToLocal(pos, lpos);
+
+  float anodpos = mMuonresponse.getAnod(lpos.X(), detID);
+
+  //TODO/Questions:
+  //- possibility to do random seeding in controlled way?
+  //distance of impact point and anode, needed for charge sharing
+  float anoddis = TMath::Abs(pos.X() - anodpos);
+  //should be related to electrons fluctuating out/in one/both halves (independent x)
+  float fracplane = mMuonresponse.chargeCorr();
+  //should become a function of anoddis
+  //std::cout << "fracplane " << fracplane << std::endl;
+  float chargebend = fracplane * charge;
+  float chargenon = charge / fracplane;
+  //last line  from Aliroot, not understood why
+  //since charge = charchbend+chargenon and not multiplication
+  float signal = 0.0;
+
+  //borders of charge gen.
+  double xMin = anodpos - mMuonresponse.getQspreadX() * 0.5;
+  double xMax = anodpos + mMuonresponse.getQspreadX() * 0.5;
+  double yMin = lpos.Y() - mMuonresponse.getQspreadY() * 0.5;
+  double yMax = lpos.Y() + mMuonresponse.getQspreadY() * 0.5;
+
+  //pad-borders
+  float xmin = 0.0;
+  float xmax = 0.0;
+  float ymin = 0.0;
+  float ymax = 0.0;
+
+  //use DetectorID to get area for signal induction
+  //single pad as check
+  int padidbendcent = 0;
+  int padidnoncent = 0;
+  bool padexists = mSeg[indexID].findPadPairByPosition(anodpos, lpos.Y(), padidbendcent, padidnoncent);
+  if (!padexists)
+    return 0; //to be decided if needed
+
+  //need to keep both electros separated since afterwards signal generation on each plane
+  //how misalignment enters?
+  std::vector<int> padIDsbend;
+  std::vector<int> padIDsnon;
+
+  //retrieve pads with signal
```

Comment:
I don't quite get here why you're treating the two cathodes (completely) separately. With the recent addition of the Segmentation class (being a pair of CathodeSegmentation), you should be able to loop on all pads of a given region of the detection element at once, shouldn't you ?

Granted, the charge sharing will induce a difference, but there is (or there should be if not yet there) a trivial way to know if a padid is from bending on non-bending.

Reply:
Well, basically because of the charge charing and the signal is hence induced on two separate areas. I would find it more convenient to separate and to loop over the pads of one area separately. 
Maybe we can discuss this and do in the next PR?